runners/java-fn-execution/src/main/java/org/apache/beam/runners/fnexecution/environment/SingletonDockerEnvironmentManager.java

```diff
@@ -82,18 +78,7 @@ private SingletonDockerEnvironmentManager(
    */
   @Override
   public RemoteEnvironment getEnvironment(Environment environment) throws Exception {
-    if (dockerEnvironment == null) {
-      dockerEnvironment = createDockerEnv(environment);
-    } else {
-      checkArgument(
-          environment.getUrl().equals(dockerEnvironment.getEnvironment().getUrl()),
-          "A %s must only be queried for a single %s. Existing %s, Argument %s",
-          SingletonDockerEnvironmentManager.class.getSimpleName(),
-          Environment.class.getSimpleName(),
-          dockerEnvironment.getEnvironment().getUrl(),
-          environment.getUrl());
-    }
-    return dockerEnvironment;
+    return createDockerEnv(environment);
```

Comment:
Again, this will need to be reworked. At this point, this is no longer an environment "manager", but a factory of sorts.

Reply:
But we need to revert this change in the future, because we can't be spinning up docker containers left and right... 😅